### PR TITLE
Ads7844 a2d - Oxygen a2d

### PR DESCRIPTION
**Pull request by JonBruchim**

### CODE REVIEW

File: algo.py [hunk 1]
```diff
@@ -12,14 +12,16 @@
 class Sampler(threading.Thread):
     SAMPLING_INTERVAL = 0.02  # sec
     MS_IN_MIN = 60 * 1000
+    OXYGEN_A2D_SAMPLE_CHANNELS = [0]
```
Comment (rinslow) [inline algo.py]: Move this to the Driver class, and add a one-line documentation explaining what this does

File: drivers/driver_factory.py [hunk 2]
```diff
@@ -122,6 +137,15 @@ def get_mock_flow_driver(self):
             data = generate_data_from_file('flow', simulation_data)
         return MockSensor(data)
 
+    def get_mock_oxygen_a2d_driver(self):
+        from drivers.mocks.sensor import MockSensor
+        simulation_data = self.simulation_data
+        if simulation_data == 'sinus':
+            data = self.generate_mock_oxygen_a2d_data()
+        else:
+            data = generate_data_from_file('oxygen', simulation_data)
```
Comment (rinslow) [inline drivers/driver_factory.py]: where is this file?

File: drivers/driver_factory.py [hunk 3]
```diff
@@ -122,6 +137,15 @@ def get_mock_flow_driver(self):
             data = generate_data_from_file('flow', simulation_data)
         return MockSensor(data)
 
+    def get_mock_oxygen_a2d_driver(self):
+        from drivers.mocks.sensor import MockSensor
```
Comment (rinslow) [inline drivers/driver_factory.py]: you are not using this import

File: drivers/mocks/mock_ads7844_a2d.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+
+class MockAds7844A2D(object):
+    sample_interval = 0.1
+
+    def __init__(self):
+        self.sample_x = 0
+        pass
+
+    def read_pressure(self):
```
Comment (rinslow) [inline drivers/mocks/mock_ads7844_a2d.py]: this does oxygen or pressure?

File: drivers/mocks/mock_ads7844_a2d.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        value = sin(self.sample_x) * 10
+        if value < 0:
+            value = 0
+
+        self.sample_x += self.sample_interval
+
+        # if 3.1 <= self.sample_x <= 3.3:
+        #     return 200
+        return value + 3
```
Comment (rinslow) [inline drivers/mocks/mock_ads7844_a2d.py]: shouldn't regular oxygen levels fluctuate between 20-22%?

File: algo.py (hunk 1 above)
Comment (Reznic) [reply algo.py]: Indeed

Comment (JonBruchim) [reply algo.py]: A2D has 8 channels that we could measure.
Currently we are measuring channel 0 only
I don't think that it needs to be a part of the driver.

File: drivers/mocks/mock_ads7844_a2d.py (hunk 4 above)
Comment (JonBruchim) [reply drivers/mocks/mock_ads7844_a2d.py]: deleting this anyway

File: drivers/driver_factory.py (hunk 2 above)
Comment (JonBruchim) [reply drivers/driver_factory.py]: we still don't have any file containing oxygen levels data
Will generate one when we will have the real oxygen sensor

File: algo.py (hunk 1 above)
Comment (rinslow) [reply algo.py]: but it is driver logic, not sampler logic or algorithmics

Comment (JonBruchim) [reply algo.py]: I think its a part of the application.
Driver should support dynamic channels.

Anyway, I've changed it to be hardcoded to the driver

File: drivers/driver_factory.py (hunk 3 above)
Comment (JonBruchim) [reply drivers/driver_factory.py]: why not?

**Comment by osherdp** [comment]
closes #60 
closes #59

File: drivers/mocks/mock_ads7844_a2d.py (hunk 5 above)
Comment (JonBruchim) [reply drivers/mocks/mock_ads7844_a2d.py]: the mock outputs a range between 17-23 (20 with offset 3)

**Pull request merged**